### PR TITLE
Convert UUID primary key to string before serializing in json.

### LINKED ISSUE #4616: Chooser fails for custom UUID primary keys in snippets
This serialization here assumes that the primary key is readily serializable:

https://github.com/wagtail/wagtail/blob/4a7e4550044ce11cb68141b23ef8ca3884e1a07d/wagtail/snippets/views/chooser.py#L77-L82

However, when the primary key is a UUIDField you get a TypeError when a snippet is selected through a snippet chooser:

    TypeError: Object of type 'UUID' is not JSON serializable

So the ìtem.pk` must be converted explicitly to a string before trying to dump it with json.

### PR DESCRIPTION
**Pull request by saevarom**
Fixes #4616 .

### CODE REVIEW

**Comment by chosak** [comment]
@saevarom as far as I know SQLite should have no problem with UUIDField. Running the tests on this branch locally all pass, and the default test configuration uses SQLite. The fact that the test command exits on Travis with `exited with code -11` suggests a segfault in SQLite - [this thread](https://github.com/travis-ci/travis-ci/issues/7873) suggests that others have experienced this on Travis as well. This wouldn't explain why CircleCI failed as well but might be a clue at least.

**Comment by saevarom** [comment]
@chosak I tried running again with the file-based sqlite instead of the in-memory one as suggested in the thread you linked to. Now the CircleCI build works, but not Travis...

File: wagtail/tests/settings.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
     }
 }
 
+if DATABASES['default']['ENGINE'] == 'sql_server.pyodbc':
```
Comment (gasman) [inline wagtail/tests/settings.py]: @saevarom This should be `django.db.backends.sqlite3` I think...?

Comment (saevarom) [reply wagtail/tests/settings.py]: Woops, your right. Paste error :)

Comment (saevarom) [reply wagtail/tests/settings.py]: I'll try again

**Comment by gasman** [comment]
I think I've found the incantation to make Travis work: https://github.com/gasman/wagtail/commit/f4752d68874facbc16919b8a25830b71d287603d

Will open a rebased PR to make sure it runs on CircleCI too, and merge if all goes well.

**Comment by gasman** [comment]
Merged in 6e60469142a11ac6df041e75c3537dcc538039ae - thanks @saevarom!

**Pull request closed without merging**